doc/classes/vector.adoc

```diff
@@ -0,0 +1,264 @@
+= Vector API
+include::../../include/config.adoc[]
+:description: Documentation for the Vector class and API
+:keywords: mod, library, vector, x, y, z, pos, position
+
+TIP: The respective source code is located https://github.com/minetest/minetest/blob/master/builtin/common/vector.lua[here].
+
+== `vector` Namespace
+
+NOTE: The `vector` type used to be a simple `table` with `x`, `y`, and `z` values.
+However it has more recently been given metatable methods for convenience.
+Unless otherwise noted, the `vector` type always refers to the metatable-enhanced variety.
+Do note that functions here will accept an old-style (non-metatable) `vector`, but you cannot perform metatable operations with said `vector`.
+
+=== `vector.new(a, b, c) -> vector`
+* `a`: `number`, `vector`, or `nil`
+* `b`, `c`: `number` or `nil`
+
+If `a`, `b`, and `c` are `number`:: Returns a new `vector` where `{x = a, y = b, z = c}`
+
+NOTE: {deprecated}
+
+If `a` is a `vector`:: Returns `vector.copy(a)`
+If all parameters are `nil`:: Returns `vector.zero()`
+
+=== `vector.zero() -> vector`
+Returns a new `vector` where `{x = 0, y = 0, z = 0}`
+
+=== `vector.copy(v) -> vector`
+* `v`: `vector`
+
+Returns a new `vector` where `{x = v.x, y = v.y, z = v.z}`
+
+=== `vector.from_string(s, init) -> vector, number`
+* `s`: `string`
+* `init`: `number`
+
+Returns a new `vector` parsed from `s` using `string.match`, followed by the number of matched values (should be 3, one for each `vector` component).
```

Comment:
This is wrong. The second return value is the starting index of the leftover string. This is accomplished through the use of an empty capture group `()` in the sources.

Reply:
My bad, would have been nice to comment that on DS's review though so that I could get it right in one go.